File: iteration1/src/Models/CourseCode.java
package iteration1.src.Models;

public class CourseCode {
    private String departmentCode;
    private int courseYear;
    private String courseID;

    // complete code is the complete course code like CSE3055 and deconstructs it here
    public CourseCode(String completeCode) {


        String[] part = completeCode.split("(?<=\\D)(?=\\d)");
        this.departmentCode = part[0];
        this.courseYear = Integer.parseInt(part[1].substring(0, 1));
        this.courseID = part[1].substring(1);
    }

    public void setCourseYear(Semester semester) {
        this.courseYear = semester.getSemesterNo();
    }

    public String getDepartmentCode() {
        return departmentCode;
    }

    public int getCourseYear() {
        return courseYear;
    }

    public String getCourseID() {
        return courseID;
    }

    @Override
    public String toString() {
        StringBuilder courseCode = new StringBuilder();
        courseCode.append(departmentCode);
        courseCode.append(courseYear);
        courseCode.append(courseID);
        return courseCode.toString();
    }
}

File: python_project/src/Models/CourseCode.py
import re


class CourseCode:

    def __init__(self, courseCode):
        parts = re.split("(?<=\\D)(?=\\d)", courseCode)

        self.deptCode = parts[0]
        self.courseYear = parts[1][0]
        self.courseID = parts[1][1:]

    def __eq__(self, other):
        return self.deptCode == other.deptCode and self.courseYear == other.courseYear and self.courseID == other.courseID

    def __str__(self):
        return f"{self.deptCode}{self.courseYear}{self.courseID}"
